### PR TITLE
Falsebottom Briefcase now shoots the weapon within the false bottom of the briefcase

### PR DESCRIPTION
**Pull request by Spaghetti-bit**
## What Does This PR Do

Fixes https://github.com/ParadiseSS13/Paradise/issues/26308

Allows guns inside the falsebottom briefcase to be fired, as intended.

TO NOTE:
Having a gun inside the falsebottom affects the click melee CD, allowing for some sick combinations. Only while it has ammo though. Once the gun inside is empty... it goes back to regular melee CD. I think this is neat.

## Why It's Good For The Game

Falsebottom is 10TC and besides hiding a high-risk item or contraband.... it's useless. It is also intended to shoot the weapon inside
> A modified briefcase capable of **storing and firing a gun** under a false bottom. Use a screwdriver to pry away the false bottom and make modifications. Distinguishable upon close examination due to the added weight.

## Images of changes

image

## Testing

Local server.

Autotot'd, bought a briefcase, a stetty, and a suppressor. Shot someone (worme) from range. 007 style.
Bought another briefcase, put a u-ion in it and performed a nasty assassination.
Bought a THIRD briefcase, put a stetty in it and bashed a worme's face in... causing the gun inside to introduce the vile creature to god.

## Changelog
:cl:
fix: Fixed the falsebottom briefcase, allowing guns within the false bottom of the briefcase to be fired again. You can now cosplay as Agent 007 and smash people in the face and shoot them simultaneously.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/firing.dm [hunk 1]
```diff
@@ -38,6 +38,12 @@
 
 /obj/item/ammo_casing/proc/throw_proj(atom/target, turf/targloc, mob/living/user, params, spread, atom/firer_source_atom)
 	var/turf/curloc = get_turf(firer_source_atom)
+	if(!istype(curloc)) // False-bottomed briefcase check.
+	{
+		var/obj/item/holding = user.get_active_hand();
+		if(istype(holding, /obj/item/storage/briefcase/false_bottomed))
+			curloc = get_turf(holding);
+	}
```
Comment (DGamerL) [inline code/modules/projectiles/firing.dm:46]: Please don't use curly brackets and `;`, this isn't C :p

Comment (DGamerL) [reply code/modules/projectiles/firing.dm:46]: (Yes DM is C derivative but shhhhh)

File: code/modules/projectiles/firing.dm [hunk 2]
```diff
@@ -38,6 +38,10 @@
 
 /obj/item/ammo_casing/proc/throw_proj(atom/target, turf/targloc, mob/living/user, params, spread, atom/firer_source_atom)
 	var/turf/curloc = get_turf(firer_source_atom)
+	if(!istype(curloc)) // False-bottomed briefcase check.
+		var/obj/item/holding = user.get_active_hand()
+		if(istype(holding, /obj/item/storage/briefcase/false_bottomed))
+			curloc = get_turf(holding)
```
Comment (DGamerL) [inline code/modules/projectiles/firing.dm:44]: When inside the briefcase, what is the `atom/firer_source_atom` here?

**Review by DGamerL — COMMENTED** [review]
> Having a gun inside the falsebottom affects the click melee CD, allowing for some sick combinations. Only while it has ammo though. Once the gun inside is empty... it goes back to regular melee CD. I think this is neat.

Can you explain this a bit further? Does the weapon hit a lot faster/have barely any cooldown?

File: code/modules/projectiles/firing.dm (hunk 2 above)
Comment (Spaghetti-bit) [reply code/modules/projectiles/firing.dm:44]: the source atom is still the stetchkin. It's just is null when inside the briefcase.

**Comment by Spaghetti-bit** [comment]
> > Having a gun inside the falsebottom affects the click melee CD, allowing for some sick combinations. Only while it has ammo though. Once the gun inside is empty... it goes back to regular melee CD. I think this is neat.
> 
> Can you explain this a bit further? Does the weapon hit a lot faster/have barely any cooldown?

It hits slightly faster, but there is still a cooldown, the same as the gun itself inside.

**Review by DGamerL — APPROVED** [review]
Seems fine

**Pull request merged**